packaging/common/rabbitmq-server-ha.ocf

```diff
@@ -0,0 +1,1585 @@
+                rc=$?
+                if [[ $rc == 0 ]] ; then
+                   ocf_log info "${LH} node '${nodename}' unjoined succesfully."
+                else
+                   ocf_log warn "${LH} unjoining node '${nodename}' failed."
+                fi
+            fi
+        done
+    done
+    return $OCF_SUCCESS
+}
+
+# Stop RMQ server process. Returns OCS_SUCCESS
+stop_server_process() {
+    local pid
+    local rc=$OCF_ERR_GENERIC
+    local LH="${LL} stop_server_process():"
+
+    pid=$(cat ${OCF_RESKEY_pid_file})
+    rc=$?
+    if [[ $rc != 0 ]] ; then
+      ocf_log err "${LH} RMQ-server process PIDFILE was not found!"
+      su_rabbit_cmd "${OCF_RESKEY_ctl} stop 2>&1 >> \"${OCF_RESKEY_log_dir}/shutdown_log\""
+      rc=$?
+      if [[ $rc == 0 ]] ; then
+          ocf_log info "${LH} RMQ-server process stopped succesfully, although there was no PIDFILE found."
+          return $OCF_SUCCESS
+      else
+          ocf_log err "${LH} Cannot stop RMQ-server process, and cannot kill it by unknown PID! Try to stop it manually!"
+          return $OCF_ERR_GENERIC
+      fi
+    fi
+
+    if [[ -z ${pid} ]] ; then
+        kill_rmq_and_remove_pid
+        return $OCF_ERR_GENERIC
+    fi
+
+    ocf_log info "${LH} Execute stop with timeout: ${TIMEOUT_ARG}"
+    su_rabbit_cmd "${OCF_RESKEY_ctl} stop ${OCF_RESKEY_pid_file} 2>&1 >> \"${OCF_RESKEY_log_dir}/shutdown_log\""
+    rc=$?
+    if [[ $rc == 0 ]] ; then
+        ocf_log info "${LH} RMQ-server process (PID=${pid}) stopped succesfully."
+    fi
+
+    kill_rmq_and_remove_pid
+    return $OCF_SUCCESS
+}
+
+# Stop RMQ-app. Return OCF_SUCCESS, if the app was stopped,
+# otherwise return OCF_ERR_GENERIC
+stop_rmq_server_app() {
+    local rc=$OCF_ERR_GENERIC
+
+    # if the beam process isn't running, then rabbit app is stopped as well
+    get_status
+    rc=$?
+    if [[ $rc != 0 ]] ; then
+        return $OCF_SUCCESS
+    fi
+
+    # stop the app
+    ocf_log info "${LH} Execute stop_app with timeout: ${TIMEOUT_ARG}"
+    su_rabbit_cmd "${OCF_RESKEY_ctl} stop_app 2>&1 >> \"${OCF_RESKEY_log_dir}/shutdown_log\""
+    rc=$?
+    if [[ $rc != 0 ]] ; then
+         ocf_log err "${LH} RMQ-server app cannot be stopped."
+         return $OCF_ERR_GENERIC
+    fi
+
+    get_status rabbit
+    rc=$?
+    if [[ $rc != $OCF_SUCCESS ]] ; then
+        ocf_log info "${LH} RMQ-server app stopped succesfully."
+        rc=$OCF_SUCCESS
+    else
+        ocf_log err "${LH} RMQ-server app cannot be stopped."
+        rc=$OCF_ERR_GENERIC
+    fi
+
+    return $rc
+}
+
+start_beam_process() {
+    local rc=$OCF_ERR_GENERIC
+    local ts_end
+    local pf_end
+    local pid
+    local LH="${LL} start_beam_process():"
+
+    # remove old PID-file if it exists
+    if [[ -f $OCF_RESKEY_pid_file ]] ; then
+        ocf_log warn "${LH} found old PID-file '${OCF_RESKEY_pid_file}'."
+        pid=$(cat ${OCF_RESKEY_pid_file})
+        if [[ -d /proc/${pid} && ! -z ${pid} ]] ; then
```

Comment:
Style issue: `! -z $pid` could be simplified as just `"$pid"`.

Also, it would be more logical to test it before the directory.